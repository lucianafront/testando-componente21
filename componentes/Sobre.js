function Sobre() {
  return (

<div className="container">
    <div className="nav-container" style={{paddingTop: "20px", paddingBottom: "20px"}}>
        <div className="brand" style={{paddingTop: "20px", paddingBottom: "20px"}}>
            <a href="https://sodiedoces.com.br/"><img
                    src="https://sodiedoces.com.br/wp-content/themes/sodiedoces/img/logo-sodie-rodape.png" alt=""/></a>
        </div>
        <nav className="nav-footer">
            <ul className="nav-list">
                <li><a href="https://sodiedoces.com.br/sobre">A Sodiê</a></li>
                <li className="show-for-large"><a id="abrirsub" href="#!">Produtos <i className="fas fa-caret-up"
                            aria-hidden="true"></i> </a>
                    <ul className="nav-dropdown">
                        <li><a href="https://sodiedoces.com.br/bolos">Bolos</a></li>
                        <li><a href="https://sodiedoces.com.br/salgados">Salgados</a></li>
                        <li><a href="https://sodiedoces.com.br/kit-festa">Kit Festa</a></li>
                        <li><a href="https://sodiedoces.com.br/doces">Doces</a></li>
                        <li><a href="https://sodiedoces.com.br/balas-de-coco">Balas de Coco</a></li>
                    </ul>
                </li>
                <li className="show-for-small"><a href="https://sodiedoces.com.br/bolos">Bolos</a></li>
                <li className="show-for-small"><a href="https://sodiedoces.com.br/salgados">Salgados</a></li>
                <li className="show-for-small"><a href="https://sodiedoces.com.br/kit-festa">Kit Festa</a></li>
                <li className="show-for-small"><a href="https://sodiedoces.com.br/doces">Doces</a></li>
                <li className="show-for-small"><a href="https://sodiedoces.com.br/balas-de-coco">Balas de Coco</a></li>
                <li><a href="https://sodiedoces.com.br/lojas">Lojas</a></li>
                <li><a href="https://sodiedoces.com.br/contato">Contato</a></li>
                <li><a href="https://sodiedoces.com.br/politica-de-privacidade" target="_blank">Política de
                        Privacidade</a></li>
            </ul>
        </nav>
        <div className="brand2 center">
            <a href="https://sodiedoces.com.br/imprensa" className="nav-franqueado center btn-marrom"
                role="button">Imprensa</a>
           
            <a href="https://suafranquia.sodiedoces.com.br/oportunidade/" target="_blank"
                className="nav-franqueado center btn-marrom" role="button">Seja um franqueado</a>
            <ul className="nav-social ulflex">
                <li>
                    <a href="https://instagram.com/SodieDoces" target="_blank" rel="noopener">
                        <i className="fab fa-instagram" aria-hidden="true"></i></a>
                </li>
                <li>
                    <a href="https://www.facebook.com/sodiedocesoficial" target="_blank" rel="noopener">
                        <i className="fab fa-facebook-square" aria-hidden="true"></i></a>
                </li>
            </ul>
        </div>
    </div>
</div>
  );
}

export default Sobre;