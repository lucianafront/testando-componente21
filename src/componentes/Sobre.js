
import Botao from "./Botao";


function Sobre() {
    return (

        <div className="container">
            <div className="nav-container" style={{ paddingTop: "20px", paddingBottom: "20px" }}>
                <div className="brand" style={{ paddingTop: "20px", paddingBottom: "20px" }}>
                    <Botao
                        link='ttps://sodiedoces.com.br/'
                    />

                    {/* <a href="https://sodiedoces.com.br/"><img */}
                    {/* src="https://sodiedoces.com.br/wp-content/themes/sodiedoces/img/logo-sodie-rodape.png" alt=""/></a> */}
                </div>
                <nav className="nav-footer">
                    <ul className="nav-list">
                        <li><a href="https://sodiedoces.com.br/sobre">A Sodiê</a></li>
                        <li className="show-for-large"><a id="abrirsub" href="#!">Produtos <i className="fas fa-caret-up"
                            aria-hidden="true"></i> </a>
                            <ul className="nav-dropdown">
                                <li><a href="https://sodiedoces.com.br/bolos">Bolos</a></li>
                                <li><a href="https://sodiedoces.com.br/salgados">Salgados</a></li>
                                <li><a href="https://sodiedoces.com.br/kit-festa">Kit Festa</a></li>
                                <li><a href="https://sodiedoces.com.br/doces">Doces</a></li>
                                <li><a href="https://sodiedoces.com.br/balas-de-coco">Balas de Coco</a></li>
                            </ul>
                        </li>
                        <li className="show-for-small"><a href="https://sodiedoces.com.br/bolos">Bolos</a></li>
                        <li className="show-for-small"><a href="https://sodiedoces.com.br/salgados">Salgados</a></li>
                        <li className="show-for-small"><a href="https://sodiedoces.com.br/kit-festa">Kit Festa</a></li>
                        <li className="show-for-small"><a href="https://sodiedoces.com.br/doces">Doces</a></li>
                        <li className="show-for-small"><a href="https://sodiedoces.com.br/balas-de-coco">Balas de Coco</a></li>
                        <li><a href="https://sodiedoces.com.br/lojas">Lojas</a></li>
                        <li><a href="https://sodiedoces.com.br/contato">Contato</a></li>
                        <li><a href="https://sodiedoces.com.br/politica-de-privacidade" target="_blank">Política de
                            Privacidade</a></li>
                    </ul>
                </nav>
                <div className="brand2 center">
                    <Botao
                        link='https://sodiedoces.com.br/imprensa'
                        id='nav-franqueado center btn-marrom'
                    />


                    <a href="https://sodiedoces.com.br/imprensa" className="nav-franqueado center btn-marrom"
                        role="button">Imprensa</a>



                    <a href="https://suafranquia.sodiedoces.com.br/oportunidade/" target="_blank"
                        className="nav-franqueado center btn-marrom" role="button">Seja um franqueado</a>
                    <ul className="nav-social ulflex">
                        <li>
                            <Botao
                                link='https://instagram.com/SodieDoces'
                                id='fab fa-instagram'
                            />


                        </li>
                        <li>
                            <Botao
                                link='https://www.facebook.com/sodiedocesoficial'
                                id='fab fa-facebook-square'
                            />

                        </li>
                    </ul>
                </div>
            </div>
        </div>
    );
}

export default Sobre;