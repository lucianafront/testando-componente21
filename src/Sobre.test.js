import React from 'react';
import { render } from '@testing-library/react';
import Sobre from './componentes/Sobre';

test('renders Sobre component correctly', () => {
  const { asFragment } = render(<Sobre />);
  expect(asFragment()).toMatchSnapshot();
});

test('calls the correct functions on link clicks', () => {
    // Mocking the functions
    window.open = jest.fn();
  
    const { getByText } = render(<Sobre />);
  
    // Simulating link clicks
    fireEvent.click(getByText('A Sodiê'));
    fireEvent.click(getByText('Produtos'));
    fireEvent.click(getByText('Bolos'));
    fireEvent.click(getByText('Salgados'));
    fireEvent.click(getByText('Kit Festa'));
    fireEvent.click(getByText('Doces'));
    fireEvent.click(getByText('Balas de Coco'));
    fireEvent.click(getByText('Lojas'));
    fireEvent.click(getByText('Contato'));
    fireEvent.click(getByText('Política de Privacidade'));
    fireEvent.click(getByText('Imprensa'));
    fireEvent.click(getByText('Seja um franqueado'));
  
    // Checking if the functions were called
    expect(window.open).toHaveBeenCalledTimes(12);
  });
