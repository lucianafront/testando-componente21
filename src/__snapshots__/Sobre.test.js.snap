// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Sobre component correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="nav-container"
      style="padding-top: 20px; padding-bottom: 20px;"
    >
      <div
        class="brand"
        style="padding-top: 20px; padding-bottom: 20px;"
      >
        <a
          href="ttps://sodiedoces.com.br/"
          rel="noreferrer"
          target="_blank"
        >
          <i
            aria-hidden="true"
          />
        </a>
      </div>
      <nav
        class="nav-footer"
      >
        <ul
          class="nav-list"
        >
          <li>
            <a
              href="https://sodiedoces.com.br/sobre"
            >
              A Sodiê
            </a>
          </li>
          <li
            class="show-for-large"
          >
            <a
              href="#!"
              id="abrirsub"
            >
              Produtos 
              <i
                aria-hidden="true"
                class="fas fa-caret-up"
              />
               
            </a>
            <ul
              class="nav-dropdown"
            >
              <li>
                <a
                  href="https://sodiedoces.com.br/bolos"
                >
                  Bolos
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/salgados"
                >
                  Salgados
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/kit-festa"
                >
                  Kit Festa
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/doces"
                >
                  Doces
                </a>
              </li>
              <li>
                <a
                  href="https://sodiedoces.com.br/balas-de-coco"
                >
                  Balas de Coco
                </a>
              </li>
            </ul>
          </li>
          <li
            class="show-for-small"
          >
            <a
              href="https://sodiedoces.com.br/bolos"
            >
              Bolos
            </a>
          </li>
          <li
            class="show-for-small"
          >
            <a
              href="https://sodiedoces.com.br/salgados"
            >
              Salgados
            </a>
          </li>
          <li
            class="show-for-small"
          >
            <a
              href="https://sodiedoces.com.br/kit-festa"
            >
              Kit Festa
            </a>
          </li>
          <li
            class="show-for-small"
          >
            <a
              href="https://sodiedoces.com.br/doces"
            >
              Doces
            </a>
          </li>
          <li
            class="show-for-small"
          >
            <a
              href="https://sodiedoces.com.br/balas-de-coco"
            >
              Balas de Coco
            </a>
          </li>
          <li>
            <a
              href="https://sodiedoces.com.br/lojas"
            >
              Lojas
            </a>
          </li>
          <li>
            <a
              href="https://sodiedoces.com.br/contato"
            >
              Contato
            </a>
          </li>
          <li>
            <a
              href="https://sodiedoces.com.br/politica-de-privacidade"
              target="_blank"
            >
              Política de Privacidade
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="brand2 center"
      >
        <a
          href="https://sodiedoces.com.br/imprensa"
          rel="noreferrer"
          target="_blank"
        >
          <i
            aria-hidden="true"
          />
        </a>
        <a
          class="nav-franqueado center btn-marrom"
          href="https://sodiedoces.com.br/imprensa"
          role="button"
        >
          Imprensa
        </a>
        <a
          class="nav-franqueado center btn-marrom"
          href="https://suafranquia.sodiedoces.com.br/oportunidade/"
          role="button"
          target="_blank"
        >
          Seja um franqueado
        </a>
        <ul
          class="nav-social ulflex"
        >
          <li>
            <a
              href="https://instagram.com/SodieDoces"
              rel="noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
              />
            </a>
          </li>
          <li>
            <a
              href="https://www.facebook.com/sodiedocesoficial"
              rel="noreferrer"
              target="_blank"
            >
              <i
                aria-hidden="true"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
